cmd/kube-apiserver/app/server.go

```diff
@@ -228,6 +228,8 @@ func CreateNodeDialer(s completedServerRunOptions) (tunneler.Tunneler, *http.Tra
 	if len(s.SSHUser) > 0 {
 		// Get ssh key distribution func, if supported
 		var installSSHKey tunneler.InstallSSHKey
+
+		cloudprovider.DeprecationWarningForProvider(s.CloudProvider.CloudProvider)
```

Comment:
ssh user is only used for a cloud provider needing ssh tunnels (i.e. gke) but that doesn't mean that the cloud provider was only set on the KAS for GKE. I think it would be better to put this warning outside the ssh user check but inside a cloud provider not external check. If we want to warn about not using ssh tunnels, I am fine with that as well but think we should be explicitly warning about ssh tunnels in that case.

Reply:
^ yes and calling it from InitCloudProvider was problematic because cloud-controller-manager also calls it, which means external providers using the same provider name (aws, gce, etc) are logging this deprecation warning incorrectly.

Reply:
I'm all fo hoisting the warning out of InitCloudProvider, I'm all for having the warning in the Kube APIServer. I just think the warning about having the cloud provider flag should be independent of if you had the SSH User flag on. If we want a separate warning for SSH User, that sounds like a good idea. However you can turn on the cloud provider flag without turning on the ssh user flag.

Reply:
Gonna fix this in a follow-up PR